### ISSUE TITLE
Allow running atlantis with github app instead of a user token

### ISSUE THREAD

**Issue by dgokcin** [terraform-aws-modules/terraform-aws-atlantis#271]
## Is your request related to a problem? Please describe.

Since we want to use atlantis in a GitHub organization, I want to have this environment variables and secrets in my atlantis container without any github user tokens.:

```terraform
  custom_environment_variables = [
    {
      name  = "ATLANTIS_GH_APP_ID"
      value = local.github_app_id
    },
    {
      "name" : "ATLANTIS_WRITE_GIT_CREDS",
      "value" : "true",
    }
  ]

  custom_environment_secrets = [
    {
      name      = "ATLANTIS_GH_APP_KEY"
      valueFrom = local.atlantis_github_app_key_ssm_parameter_name
    }
  ]
```
The module currently depends on the presence of `var.atlantis_github_user_token`, which is used as a conditional in the creation of some resources and that brings a problem:
- Without the atlantis_github_user_token, although there may be other secrets in the environment(webhook_secret. github app key secret etc.), the tasks can not get their value.
https://github.com/terraform-aws-modules/terraform-aws-atlantis/blob/fab764c06444a0e72deb6c3cf92bdb4fbc31ba4b/main.tf#L567

I worked around this issue by modifying the module by adding some custom variables.

## Describe the solution you'd like.

I would like to create the atlantis module with a simple parameter like `use_github_bot = true` which will take care of setting up all the necessary environment variables/parameter store secrets for me.

**Comment by dgteixeira** [comment]
Hey, I know this is a recent issue, however I want to add myself to the list of interested parties ! 😋

**Comment by antonbabenko** [comment]
This issue has been resolved in version 3.23.0 :tada:
